resources/appmesh-mesh.go

```diff
@@ -0,0 +1,64 @@
+package resources
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/appmesh"
+	"github.com/rebuy-de/aws-nuke/pkg/types"
+)
+
+type AppMeshMesh struct {
+	svc      *appmesh.AppMesh
+	meshName *string
+}
+
+func init() {
+	register("AppMeshMesh", ListAppMeshMeshes)
+}
+
+func ListAppMeshMeshes(sess *session.Session) ([]Resource, error) {
+	svc := appmesh.New(sess)
+	resources := []Resource{}
+
+	params := &appmesh.ListMeshesInput{
+		Limit: aws.Int64(100),
+	}
+
+	for {
+		output, err := svc.ListMeshes(params)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, item := range output.Meshes {
+			resources = append(resources, &AppMeshMesh{
+				svc:      svc,
+				meshName: item.MeshName,
+			})
+		}
+
+		if output.NextToken == nil {
+			break
+		}
+
+		params.NextToken = output.NextToken
+	}
+
+	return resources, nil
+}
+
+func (f *AppMeshMesh) Remove() error {
+	_, err := f.svc.DeleteMesh(&appmesh.DeleteMeshInput{
+		MeshName: f.meshName,
+	})
+
+	return err
+}
+
+func (f *AppMeshMesh) Properties() types.Properties {
+	properties := types.NewProperties()
+	properties.
+		Set("MeshName", f.meshName)
```

Comment:
Normally we would say "Name" is enough, but this would allow to have the same filter on all "sub"resources of a AppMesh. @svenwltr @der-eismann whats your opinion here?

With "MeshName" all sub resources could be filtered the same way, with having Name here we would stay consistent with other resources, but the filter would look different.

Reply:
I'd say it should be consistent within a AWS-Service.